engine/runtime/src/main/java/org/enso/interpreter/runtime/error/PanicException.java

```diff
@@ -231,4 +282,15 @@ private static Logger logger() {
     CompilerDirectives.transferToInterpreter();
     return LoggerFactory.getLogger(PanicException.class);
   }
+
+  /**
+   * Whoever catches a {@code PanicException} should associate its location with it, so proper stack
+   * traces can be computed later. Because the stacktrace information is relative to the caught
+   * location
+   *
+   * @param node the node who caught the panic
+   */
+  public final void assignCaughtLocation(CatchPanicNode node) {
+    this.caughtBy = node;
+  }
```

Comment:
What happens if the panic is rethrown and caught again elsewhere? Should this keep the first catch location or be reassigned?

Reply:
That's a very good question. It points out that this all is _just a heuristic_ with the goal (as stated in the description):

> ... fix the most common scenario when a `caught_exception.stack_trace` is requested in the exception handler

I will add a new 
- test with re-throw.

>  Should this ... be reassigned?

Both `CatchPanicNode` and `Node queryNode` have to be on the stack for any transformation to happen. Thus  yes, it has to be reassigned. 

I will also add a test that
- catches the panic
- queries the `stack_trace` in the handler
- stores the panic into a `Vector`
- and then queries later
- both stacks have to be identical